app/uk/gov/hmrc/exports/models/Pointer.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.models
+
+import play.api.libs.json._
+import uk.gov.hmrc.exports.models
+import uk.gov.hmrc.exports.models.PointerSectionType.PointerSectionType
+
+import scala.util.{Success, Try}
+
+object PointerSectionType extends Enumeration {
+  type PointerSectionType = Value
+  val FIELD, SEQUENCE = Value
+  implicit val format: Format[models.PointerSectionType.Value] =
+    Format(Reads.enumNameReads(PointerSectionType), Writes.enumNameWrites)
+}
+
+case class PointerSection(value: String, `type`: PointerSectionType) {
+  lazy val pattern: String = `type` match {
+    case PointerSectionType.FIELD    => value
+    case PointerSectionType.SEQUENCE => "$"
+  }
+}
+
+object PointerSection {
+  implicit val format = Json.format[PointerSection]
+}
+
+case class Pointer(sections: Seq[PointerSection]) {
+  //  Converts a pointer into it's pattern form
+  // e.g. ABC.DEF.*.GHI (if the pointer contains a sequence index)
+  // e.g. ABC.DEF.GHI (if the pointer doesnt contain a sequence)
+  lazy val pattern: PointerPattern = PointerPattern(sections.map(_.pattern).mkString("."))
+
+  // Converts a pointer into its value form
+  // e.g. ABC.DEF.1.GHI  (if the pointer contains a sequence with index 1)
```

Comment:
Is it possible to have  `ABC.DEF.2.GHI` or  `ABC.DEF.4.GHI`?
In what way would the field differ from a `ABC.DEF.GHI`?
Would it just be to point to the user which element is causing an issue?